src/http_server.cpp

```diff
@@ -315,6 +322,10 @@ namespace sim
 		m_recv_buffer.erase(m_recv_buffer.begin(), m_recv_buffer.begin() + req_len);
 		m_bytes_used -= req_len;
 
+		if (m_send_buffer.empty()) {
```

Comment:
this is a bit problematic, because it conflates an empty response with no response. ideally there would be a way for a handler to indicate that it wanted to suppress the response altogether, or perhaps that should be the behavior of there is no handler registered, or something else, separate from the regular handlers